src/models/plot.ts

```diff
@@ -77,3 +76,30 @@ function plotObject(data: Data, specQ: SpecQueryModel): PlotObject {
 
   return {fieldInfos, spec};
 }
+
+function toShelfFieldDef(fieldQ: FieldQuery): ShelfFieldDef {
```

Comment:
This is a common utility for shelf field def, put it in model and name it `fromFieldQuery`?